TST: test_no_numpy_warnings failed in main branch CI due to matplotlib 3.4.3

This might block unrelated PRs. Also would be nice to have it green again on `main` ASAP. I don't have time to look into this until next week, so would be nice if someone else can look into it in the meantime. It is `astropy.visualization.wcsaxes` again! And the biggest suspicion is the new [matplotlib 3.4.3 release](https://github.com/matplotlib/matplotlib/releases/tag/v3.4.3) that happened 12 hours ago.

```
_______________________ test_no_numpy_warnings[contours] _______________________

ignore_matplotlibrc = None
tmpdir = local('/tmp/pytest-of-runner/pytest-0/test_no_numpy_warnings_contour0')
grid_type = 'contours'

    @pytest.mark.parametrize('grid_type', ['lines', 'contours'])
    def test_no_numpy_warnings(ignore_matplotlibrc, tmpdir, grid_type):
        ax = plt.subplot(1, 1, 1, projection=WCS(TARGET_HEADER))
        ax.imshow(np.zeros((100, 200)))
        ax.coords.grid(color='white', grid_type=grid_type)
    
        if MATPLOTLIB_GT_3_4_2 and grid_type == 'contours':
            ctx = pytest.raises(AttributeError, match='dpi')
        else:
            ctx = nullcontext()
    
        with pytest.warns(None) as warning_lines, ctx:
>           plt.savefig(tmpdir.join('test.png').strpath)
E           Failed: DID NOT RAISE <class 'AttributeError'>

../../astropy/visualization/wcsaxes/tests/test_misc.py:88: Failed
```

--- a/astropy/visualization/wcsaxes/tests/test_misc.py
+++ b/astropy/visualization/wcsaxes/tests/test_misc.py
@@ -25,7 +25,7 @@ ft_version = Version(matplotlib.ft2font.__freetype_version__)
 FREETYPE_261 = ft_version == Version("2.6.1")
 TEX_UNAVAILABLE = not matplotlib.checkdep_usetex(True)
 
-MATPLOTLIB_GT_3_4_2 = Version(matplotlib.__version__) > Version('3.4.2')
+MATPLOTLIB_GT_3_4_3 = Version(matplotlib.__version__) > Version('3.4.3 ')
 
 
 def teardown_function(function):
@@ -79,7 +79,7 @@ def test_no_numpy_warnings(ignore_matplotlibrc, tmpdir, grid_type):
     ax.imshow(np.zeros((100, 200)))
     ax.coords.grid(color='white', grid_type=grid_type)
 
-    if MATPLOTLIB_GT_3_4_2 and grid_type == 'contours':
+    if MATPLOTLIB_GT_3_4_3 and grid_type == 'contours':
         ctx = pytest.raises(AttributeError, match='dpi')
     else:
         ctx = nullcontext()

